bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -2923,23 +2923,36 @@ private static boolean hasFillerValue(Type type, List<Type> unanalyzedTypes) {
             case TypeTags.ANY_TAG:
                 return true;
             case TypeTags.ARRAY_TAG:
-                return checkFillerValue((BArrayType) type);
+                return checkFillerValue((BArrayType) type, unanalyzedTypes);
```

Comment:
Not related to this PR, but we should probably be using a set instead of a list for unanalyzedTypes. Can you please create an issue for this?

Reply:
#31894